SDPhysicsEngine/Sources/SDPhysicsEngine/GameScene.swift

```diff
@@ -81,17 +74,26 @@ extension GameScene: SDScene {
         object.removeFromParent()
     }
 
-    private func playerScreenSize(for numberOfPlayers: Int) -> CGSize {
+    private func scaledPlayerScreenSize(for numberOfPlayers: Int) -> CGSize {
```

Comment:
GameScene ideally should be blind to the layout of the screen, or at least the size of the rendered view should not be assumed here (it should be received from some source like the renderer). This leads to some hidden coupling between the renderer and the game scene.

Reply:
Mm I understand. I agree that the `Renderer` should be the one dealing with the rendered screen's size.

My changes currently refactor the flow of updating the camera constraints like this:
1. When `GameBridge` needs to create a new player object, it calls `GameScene::addPlayerObject`
2. `GameScene::addPlayerObject` creates a basic camera with the player, but without any of the constraints and scale
3. `GameScene::addPlayerObject` delegates setting up the camera constraints and scale to its `sceneDelegate` (i.e. `GameViewController::setupCameras`)
4. `GameViewController::setupCameras` calls `MTKRenderer::setupCameras`, which derives the player screen size from `LayoutUtils` and calls `GameScene::setupCameras`
5. `GameScene::setupCameras`, now with the player screen size, then calls `SDCameraObject::setup` to set the constraints and camera scale

Do you think this is too convoluted?

Reply:
I think it feels a little convoluted, not too sure if the viewcontroller needs to be involved anymore... 
I think MTKRenderer can setup the constraints and scale in `updateScene` where it calls `scene.useCamera`, then `GameScene` just record if the constraints/scale has been changed to ensure it only does it once?

Reply:
good idea!